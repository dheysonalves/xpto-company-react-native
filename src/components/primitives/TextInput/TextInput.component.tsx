import React, { useState } from "react";
import { View, Text, TextInput, StyleSheet } from "react-native";

interface InputDataProps {
	title: string;
	dataValue: string;
	onHandleDataValue: (value: string) => void;
	onHandleBlur?: (value: string) => void;
	hasError: boolean;
	errorMessage?: string;
}

const InputData = ({
	title,
	dataValue,
	onHandleDataValue,
	hasError,
	errorMessage,
	onHandleBlur,
	...props
}: InputDataProps & TextInput['props']) => {
	const [isFocused, setIsFocused] = useState(false);
	const handleDataValue = (value: string) => {
		onHandleDataValue(value);
	};

	return (
		<View style={styles.container}>
			<View style={styles.wrapperText}>
				<Text style={[styles.baseText, styles.title]}>{title}</Text>
			</View>
			<View style={styles.inputContainer}>
				<TextInput
					style={[
						styles.input,
						isFocused ? styles.focused : null,
						hasError ? styles.error : null,
					]}
					selectionColor="hsl(183, 100%, 15%)"
					value={dataValue}
					onChangeText={handleDataValue}
					onFocus={() => {
						setIsFocused(true);
					}}
					onBlur={() => {
						setIsFocused(false);
						onHandleBlur;
					}}
					{...props}
				/>
			</View>
			{hasError && (
				<Text style={[styles.baseText, styles.errorText]}>{errorMessage}</Text>
			)}
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		alignItems: "flex-start",
		margin: 12,
	},
	baseText: {
		fontWeight: "700",
		letterSpacing: 0.6,
		fontSize: 18,
		marginVertical: 4,
	},
	wrapperText: {
		flexDirection: "row",
		justifyContent: "space-between",
		alignItems: "center",
		width: '100%',
	},
	title: {
		color: "hsl(184, 14%, 56%)",
	},
	inputContainer: {
		width: "100%",
	},
	input: {
		backgroundColor: "hsl(189, 41%, 97%)",
		paddingHorizontal: 16,
		paddingVertical: 8,
		borderRadius: 5,
		fontSize: 24,
		color: "hsl(183, 100%, 15%)",
		fontWeight: "700",
		letterSpacing: 0.6,
	},
	focused: {
		borderColor: "hsl(172, 67%, 45%)",
		borderWidth: 1,
	},
	errorText: {
		color: "red",
		fontSize: 14,
	},
	error: {
		borderColor: "red",
		borderWidth: 1,
	},
});

export default InputData;
