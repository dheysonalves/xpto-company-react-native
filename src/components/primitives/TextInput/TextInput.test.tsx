import React from "react";

import TextInput from "./TextInput.component";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import { fireEvent, render, waitFor, cleanup } from "@testing-library/react-native";

afterEach(cleanup);

const TextInputRender = () => {
	const [name, setName] = React.useState("");

	return (
		<>
			<TextInput
				dataValue={name}
				onHandleDataValue={setName}
				hasError={false}
				title="Input"
				testID="textInputType"
				accessibilityLabel="Type me"
			/>
			{name !== "" && <Text testID="printed-username">{name}</Text>}
		</>
	);
};

describe("Styled TextInput Test Suite", () => {
	it("should render the TextInput", () => {
		const tree = renderer.create(<TextInputRender />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should find the TextInput title", () => {
		const title = "Input";

		const { getByText } = render(<TextInputRender />);

		const foundButtonTitle = getByText(title);

		expect(foundButtonTitle.props.children).toEqual(title);
	});

	it("should type and return a text in the screen", async () => {
		const famousProgrammerInHistory = "Ada Lovelace";

		const { getByText, getByTestId, queryByTestId } = render(
			<TextInputRender />
		);

		const input = getByTestId("textInputType");
		fireEvent.changeText(input, famousProgrammerInHistory);

		await waitFor(() => expect(queryByTestId("printed-username")).toBeTruthy());
	});
});
