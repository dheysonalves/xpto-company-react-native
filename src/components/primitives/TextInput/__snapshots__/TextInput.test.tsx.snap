// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled TextInput Test Suite should render the TextInput 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "margin": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "700",
            "letterSpacing": 0.6,
            "marginVertical": 4,
          },
          Object {
            "color": "hsl(184, 14%, 56%)",
          },
        ]
      }
    >
      Input
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <TextInput
      accessibilityLabel="Type me"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      selectionColor="hsl(183, 100%, 15%)"
      style={
        Array [
          Object {
            "backgroundColor": "hsl(189, 41%, 97%)",
            "borderRadius": 5,
            "color": "hsl(183, 100%, 15%)",
            "fontSize": 24,
            "fontWeight": "700",
            "letterSpacing": 0.6,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          null,
          null,
        ]
      }
      testID="textInputType"
      value=""
    />
  </View>
</View>
`;
